locust/configuration.py

```diff
@@ -79,17 +78,96 @@ def update_json_config(self, json_added, json_path, options, list_column, config
+            return make_response(json.dumps({'success':False, 'message':'JSON path not found.', 'key_pass_status':False}))
         
         for match in matches:
             data = ClientConfiguration.update_json(data, ClientConfiguration.get_path(match), json_final)
         
-        return make_response(json.dumps({'success':True, 'data':json.dumps(data, indent=4)}))
+        if 'data' in kwargs:
+            return data
+        else:
+            return make_response(json.dumps({'success':True, 'data':json.dumps(data, indent=4), 'key_pass_status':False}))
+
+    def update_json_each_row_depend_on_key(self, json_added, json_path, options, list_column, config_text, key_json, key_csv):
+        """
+        Write JSON file configuration each data json array and csv row depends on a key
+        """ 
+        temppath = json_path.split('.')
+        if(len(temppath[len(temppath)-2]) > 0):
+            return make_response(json.dumps({'success':False, 'message':'Check again your jsonpath'}))
+        
+        data = literal_eval(config_text)
+        
+        for x in xrange(1,len(temppath)-1):
+            if(len(temppath[x]) > 0):
+                data = data[temppath[x]]
         
+        datas = None
+        missing_data_key_csv = []
+        for y in xrange(0,len(json_added)):
+            index_search = ClientConfiguration.find_index(data, key_json, json_added[y][key_csv])
+            if index_search < 0:
+                missing_data_key_csv.append(json_added[y][key_csv])
+            else:
+                json_path_iter = '.'.join(temppath[:-1]) + "[" + str(index_search) + "]." + temppath[-1]
+                if y==0:
+                    data_each_iter = literal_eval(config_text)
+                else:
+                    data_each_iter = datas
+                json_added[y].pop(key_csv)
+                input = []
+                input.append(json_added[y])
+
+                #Passing data_each_iter for every iteration. ex: iteration 1 will use data from iteration 0, iteration 2 will use data from iteration 1
+                datas = ClientConfiguration.update_json_config(self, input, json_path_iter, options, list_column, config_text, data=data_each_iter)
+
+        if len(missing_data_key_csv) == 0:
+            key_pass_status = False
+            missing_key_message = ""
+        elif len(missing_data_key_csv) == len(json_added):
+            key_pass_status = True
+            missing_key_message = "All data failed to insert"
+        else:
+            key_pass_status = True
+            missing_key_message = "Insert success but some data failed : " + ','.join(missing_data_key_csv)
+        
+        return make_response(json.dumps({'success':True, 'data':json.dumps(datas, indent=4), 'key_pass_status':key_pass_status, 'missing_key_message':missing_key_message}))
+        
+    def update_json_each_row_on_sequence(self, json_added, json_path, options, list_column, config_text):
```

Comment:
this function name is long but not quite representative since json doesn't use row. how about if we rename it to `update_configuration_join_by_row_order` ?